### PR TITLE
Feature/openai add missing usage

### PR DESCRIPTION
**Pull request by glharper**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/main.cadl [hunk 1]
```diff
@@ -48,3 +58,10 @@ model Deployment {
   @key
   deploymentId: string;
 }
+
+@doc("Tokens input/total for a given API")
+model Usage {
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/main.cadl:63]: You need one per file, as they are not the same (EmbeddingUsage and CompletionUsage). CompletionUsage will have an additional `completion_tokens: int32`.

File: specification/cognitiveservices/OpenAI.Inference/main.cadl [hunk 2]
```diff
@@ -48,3 +58,10 @@ model Deployment {
...
+    prompt_tokens: int32,
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/main.cadl:64]: Should have a `@doc` with some description

File: specification/cognitiveservices/OpenAI.Inference/main.cadl [hunk 3]
```diff
@@ -48,3 +58,10 @@ model Deployment {
...
+    total_tokens: int32
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/main.cadl:65]: Should have a `@doc` with some description

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl [hunk 4]
```diff
@@ -138,6 +138,9 @@ model Completions {
     "model"?: string;
     @doc("Array of choices returned containing text completions to prompts sent")
     choices?: Choice[];
+    @added(ServiceApiVersions.v2022_12_01)
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl:141]: This is untrue, as this field is already available in the preview API version and was not added in 2022-12-01. I would remove the line

File: specification/cognitiveservices/OpenAI.Inference/models/embeddings.create.cadl [hunk 5]
```diff
@@ -24,18 +24,17 @@ model EmbeddingsOptions {
     as we have observed inferior results when newlines are present.
     """)
     input: string | string[];
+
+    @added(ServiceApiVersions.v2022_12_01)
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/models/embeddings.create.cadl:28]: This is untrue, as this field is already available in the preview API version and was not added in 2022-12-01. I would remove the line

File: specification/cognitiveservices/OpenAI.Inference/main.cadl (hunk 1 above)
Comment (johanste) [inline specification/cognitiveservices/OpenAI.Inference/main.cadl:63]: We should include which version this model was added to (e.g. `@Versioning.added(ServiceApiVersions.v2022-12-01)`)

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl [hunk 6]
```diff
@@ -163,3 +165,14 @@ model CompletionsLogProbsModel {
     @doc("Text offset")
     text_offset?: int32[];
 }
+
+@Versioning.added(Azure.OpenAI.ServiceApiVersions.v2022_12_01)
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl:169]: There were not added in this version, as they were already in the previous one. I don't think this PR should use `added` at all actually

File: specification/cognitiveservices/OpenAI.Inference/models/embeddings.create.cadl [hunk 7]
```diff
@@ -24,20 +24,26 @@ model EmbeddingsOptions {
...
-}
-
 model EmbeddingItem {
     object: "embedding",
     embedding: float32[];
     index: int32;
 }
+
+@doc("Measurment of the amount of tokens used in this request and response")
+model Usage {
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/models/embeddings.create.cadl:44]:
```suggestion
model EmbeddingsUsage {
```

File: specification/cognitiveservices/OpenAI.Inference/models/embeddings.create.cadl [hunk 8]
```diff
@@ -24,20 +24,26 @@ model EmbeddingsOptions {
     as we have observed inferior results when newlines are present.
     """)
     input: string | string[];
+
+    @doc("Usage counts for tokens input using the embeddings API")
+    usage: Usage;
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/models/embeddings.create.cadl:29]:
```suggestion
    usage: EmbeddingsUsage;
```

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl [hunk 9]
```diff
@@ -138,6 +138,8 @@ model Completions {
     "model"?: string;
     @doc("Array of choices returned containing text completions to prompts sent")
     choices?: Choice[];
+    @doc("Usage counts for tokens input using the completions API")
+    usage: Usage;
```
Comment (lmazuel) [inline specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl:142]:
```suggestion
    usage: CompletionsUsage;
```

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl [hunk 10]
```diff
@@ -163,3 +165,14 @@ model CompletionsLogProbsModel {
     @doc("Text offset")
     text_offset?: int32[];
 }
+
+@Versioning.added(Azure.OpenAI.ServiceApiVersions.v2022_12_01)
+@doc("Measurment of the amount of tokens used in this request and response")
+model Usage {
```
[the remark this answers is not in the document]
Comment (jpalvarezl) [reply specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl:171]: are you totally sure? We are in different namespaces. We have both `Completions` and `Inference`

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl (hunk 6 above)
Comment (jpalvarezl) [reply specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl:169]: It appears we were missing it for `Completions`

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl (hunk 10 above)
[the remark this answers is not in the document]
Comment (jpalvarezl) [reply specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl:171]: Resolved offline

**Review by lmazuel — APPROVED** [review]

**Pull request merged**